rfcs/text/0000-dss-metrics.md

```diff
@@ -0,0 +1,65 @@
+### DCP PR:
+
+***Leave this blank until the RFC is approved** then the **Author(s)** must create a link between the assigned RFC number and this pull request in the format:*
+
+`[dcp-community/rfc#](https://github.com/HumanCellAtlas/dcp-community/pull/<PR#>)`
+
+# RFC: Key performance metrics of the DCP Data Store (DSS)
+
+## Summary
+
+This RFC proposes a set of key performance metrics to be introduced and tracked as a summary measure of performance by
+which the utility of the Data Store can be evaluated, including improvements and regressions.
+
+## Author(s)
+
+* [Andrey Kislyuk](mailto:akislyuk@chanzuckerberg.com)
+
+## Shepherd
+***Leave this blank.** This role is assigned by DCP PM to guide the **Author(s)** through the RFC process.*
+
+*Recommended format for Shepherds:*
+
+ `[Name](mailto:username@example.com)`
+
+## Motivation
+
+Beyond latency and error rates, the DSS lacks a clear focus on quantifiable assessment of usability. We would like to
+quantify higher level performance metrics that reflect the Data Store's ability to make HCA data accessible, computable,
+and durable.
+
+### User Stories
+
+* As a DSS developer, I would like to know whether the DSS is improving in its ability to serve downstream users and
+  services.
+
+* As a downstream service developer, I would like to suggest a way of measuring DSS performance that reflects my ability
+  to use the DSS to consume HCA data.
+
+* As a DCP funder, I would like to know how the DSS is balancing the goals of economical data storage and data
+  accessibility and computability.
+
+## Detailed Design
+
+Introduce infrastructure (such as logging, instrumentation, offline log analytics, etc.) to measure the following DSS
+performance metrics:
```

Comment:
I think overall, I don't disagree with any of the listed metrics here but I'd like to see a slightly more built out description and perhaps even more metrics. You could perhaps format them in a table where the columns are: Metric Name, Description, Units. 

Underneath the table, I would enumerate the metrics names again and bullet point under each one, the expected infrastructure changes that need to take place. E.g.

**Unrecoverable Error Rate**
- May require added instrumentation in the HCA CLI

Reply:
On second thought, you could potentially even bucket them into the three overall categories if it makes sense: 1) API Metrics, 2) Error Metrics, 3) Costing Metrics.

Reply:
Regarding the re-formatting: how does the re-formatting improve the RFC? It's hard to work with Markdown tables that contain more than a small snippet of text. While readability of the rendered text may be improved a little bit, readability of the Markdown source is drastically reduced.

Regarding expanding the descriptions: I don't mind adding more detail, but what specifically would you like to see?

I've added units to each listing, and augmented the metrics with a list of categories that you mentioned (thanks!)

Regarding adding more metrics: I suggest not overloading this RFC with too many metrics; this is intended as a starting point for KPI measurement. Teams should be free to add more metrics after the first set is implemented.

Reply:
Makes sense. I was thinking that there would be even more metrics here but with your statement that this RFC not be overloaded the re-format suggestion doesn't make sense anymore.